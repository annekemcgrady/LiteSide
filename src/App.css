* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  background-color: #3e3e3e;
  border: 10px solid #fff;
  /* border: 10px solid #ffffa7; */
  display: flex;
  height: 100vh;
  justify-content: space-around;
  text-align: center;
  width: 100vw;
  align-items: center;
}



