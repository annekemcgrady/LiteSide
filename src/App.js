import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import './App.css';
import Aside from './components/Aside/Aside.js';
import CardContainer from './components/CardContainer/CardContainer.js'
import Nav from './components/Nav/Nav.js';
import Home from './components/Home/Home.js';
import { fetchFilms, fetchPeople, fetchPlanets, fetchVehicles } from './apiCalls.js';
import loadingImage from './images/loading-y.gif'

class App extends Component {
  constructor(){
    super()
    this.state = {
      films: [],
      data: [],
      favCount: 0,
      asideShow: true,
      error: '',
    }
  }
componentDidMount() {

fetchFilms().then(films =>
    this.setState({films: films.results})
  )

  fetchPeople().then(people =>
    people.results.forEach((person, i) => {
      person.type = 'people';
      person.id = `pe-${i}`;
      person.favorite = false;
      this.setState({data: [...this.state.data, person]})
    })
  )

  fetchPlanets().then(planets =>
    planets.results.forEach((planet, i) => {
      planet.type = 'planets';
      planet.id = `pl-${i}`;
      planet.favorite = false;
      this.setState({data: [...this.state.data, planet]})
    })
  )

  fetchVehicles().then(vehicles =>
    vehicles.results.forEach((vehicle, i) => {
      vehicle.type = 'vehicles';
      vehicle.id = `vh-${i}`;
      vehicle.favorite = false;
      this.setState({data: [...this.state.data, vehicle]})
    })
  )
    
  }

  handleFavorite = (id) => {
    const items = this.state.data.map(item => {
      if (item.id === id) {
        item.favorite = !item.favorite;
      }
      return item
    });
    this.setState({data: [...items]})

    this.favCounter();

  }

  favCounter = () => {
    let numOfFavorites = 0;
     this.state.data.forEach(i => {
      if (i.favorite) {
        numOfFavorites++;
      }
    }
    )
    this.setState({favCount: numOfFavorites})
  }


   filterData = (type) => {
    if (type !== 'favorite') {
      return this.state.data.filter(item => 
        item.type === type
        )
     } else {
      return this.state.data.filter(item => 
        item.favorite === true
        )
     }
   }
   
   updateAsideShow = () => {
    this.setState({asideShow: !this.state.asideShow})
   }

  render() {

    const {films, favCount, data, asideShow} = this.state;

    return (
      <div className="App">
        {((films.length === 0) || (data.length === 0)) && <div className="loading-container">
          <img className="loading" src={loadingImage} alt="loading films"/>
          </div>}

        {(films.length !== 0) && <Aside films={films} updateAsideShow={this.updateAsideShow} asideShow={asideShow}/> }

        <Route exact path='/' component={Home} />
        <Route exact path='/people' render={({match})=> {

          return (data.length !== 0) && <CardContainer handleFavorite={this.handleFavorite} data={this.filterData('people')} path={match.path} updateCurrentPath={this.updateCurrentPath} asideShow={asideShow}/>
          }} />

        <Route exact path='/planets' render={({match})=> {
        return (data.length !== 0)  && <CardContainer handleFavorite={this.handleFavorite} data={this.filterData('planets')} path={match.path} updateCurrentPath={this.updateCurrentPath} asideShow={asideShow}/>}}/>


        <Route exact path='/vehicles' render={({match})=> {
        return (data.length !== 0) &&  <CardContainer handleFavorite={this.handleFavorite} data={this.filterData('vehicles')} path={match.path} updateCurrentPath={this.updateCurrentPath} asideShow={asideShow}/>}}/>

        <Route exact path='/favorites' render={({match})=> {
        return (data.length !== 0) && <CardContainer  handleFavorite={this.handleFavorite} data={this.filterData('favorite')} favCount={favCount} path={match.path} updateCurrentPath={this.updateCurrentPath} asideShow={asideShow}/> }} />

        <Nav favCount={favCount}/>

        
      </div>
    );
  }

}

export default App;
