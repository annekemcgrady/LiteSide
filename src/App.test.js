import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';


describe ('App', ()=> {
  let wrapper;


  beforeEach(() => {
    wrapper = shallow(<App />);
     

    
  });

it('should match the snapshot', ()=> {
  expect(wrapper).toMatchSnapshot();
  }) 

  it('should toggle favorite ', () => {
    const fakeData = [{
      birth_year: "19BBY",
      favorite: true,
      gender: "male",
      id: "pp-0",
      mass: "77",
      name: "Luke Skywalker",
      type: "people",
    }]

     const expected = [{
      birth_year: "19BBY",
      favorite: false,
      gender: "male",
      id: "pp-0",
      mass: "77",
      name: "Luke Skywalker",
      type: "people",
    }]

    wrapper.instance().setState({data: fakeData})

    wrapper.instance().handleFavorite("pp-0")
    expect(wrapper.state('data')).toEqual(expected)
  })

  it('favCounter should increment/decrement faveCount in state', () => {

    const fakeData = [{
      birth_year: "19BBY",
      favorite: true,
      gender: "male",
      id: "pp-0",
      mass: "77",
      name: "Luke Skywalker",
      type: "people",
    }]

     const expected = [{
      birth_year: "19BBY",
      favorite: false,
      gender: "male",
      id: "pp-0",
      mass: "77",
      name: "Luke Skywalker",
      type: "people",
    }]

    wrapper.instance().setState({data: fakeData})
    wrapper.instance().setState({favCount: 0})
    wrapper.instance().favCounter()
    expect(wrapper.state('favCount')).toEqual(1)
  })


  it.skip('filterData should filter data by type and/or favorite', () => {

    const fakeData = [{
      favorite: true,
      id: "pp-0",
      type: "people",
    },
    {
      favorite: true,
      id: "pp-0",
      type: "vehicles",
    },
    {
      favorite: true,
      id: "pp-0",
      type: "people",
    }
  ]

     const expected = [{
      favorite: true,
      id: "pp-0",
      type: "people",
    },
    {
      favorite: true,
      id: "pp-0",
      type: "people",
    }]

    wrapper.instance().setState({data: fakeData})

    wrapper.instance().filterData('people')

    expect(wrapper(filterData).toEqual(expected))
  })


  it('updateAsideShow should toggle asideShow in state', () => {
    wrapper.instance().setState({asideShow: true})
    wrapper.instance().updateAsideShow()
    expect(wrapper.state('asideShow')).toEqual(false)
  })




  
});

