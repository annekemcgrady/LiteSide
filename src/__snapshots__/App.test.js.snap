// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    exact={true}
    path="/people"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/planets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vehicles"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Nav
    favCount={0}
  />
</div>
`;
